lib/src/backend.rs

```diff
@@ -165,8 +165,28 @@ pub enum BackendError {
         object_type: &'static str,
         hash: String,
     },
-    #[error("Object not found")]
-    NotFound,
+    #[error("Invalid hash for object of type {object_type}: {hash}: {source}")]
+    InvalidHash {
+        object_type: &'static str,
+        hash: String,
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+    #[error("Object of type {object_type} with hash {hash} not found: {source}")]
+    ObjectNotFound {
+        object_type: &'static str,
+        hash: String,
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+    #[error("Error when reading object of type {object_type} with hash {hash}: {source}")]
+    ReadObject {
+        object_type: &'static str,
+        hash: String,
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+    #[error("Could not create commit: {source}")]
+    CreateCommit {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
```

Comment:
Maybe we'll want to make the `source` fields optional (if a backend wants to say e.g. `ObjectNotFound` without having an associated `Error` to pass it), but that can come later.